sql-to-dbsp-compiler/SQL-compiler/src/main/java/org/dbsp/sqlCompiler/compiler/frontend/ExpressionCompiler.java

```diff
@@ -995,6 +995,45 @@ node, new DBSPTypeBool(CalciteObject.EMPTY, false), DBSPOpcode.EQ,
                 }
                 return this.compileUDF(node, call, type, ops);
             }
+            case ARRAYS_OVERLAP: {
+                if (ops.size() != 2)
+                    throw new UnimplementedException(node);
+
+                DBSPExpression arg0 = ops.get(0);
+                DBSPTypeVec arg0Vec = arg0.getType().to(DBSPTypeVec.class);
+                DBSPType arg0ElemType = arg0Vec.getElementType();
+
+                DBSPExpression arg1 = ops.get(1);
+                DBSPTypeVec arg1Vec = arg1.getType().to(DBSPTypeVec.class);
+                DBSPType arg1ElemType = arg1Vec.getElementType();
+
+                // if the two arrays are of different types of elements
+                if (!arg0ElemType.sameType(arg1ElemType)) {
+                    // check if the nullability of arg0 needs to change to match arg1
+                    if (arg1ElemType.mayBeNull && !arg0ElemType.mayBeNull) {
+                        if (arg0Vec.mayBeNull && arg0.equals(arg0Vec.nullValue())) {
+                            arg0 = new DBSPTypeVec(arg1ElemType, true).nullValue();
+                            ops.set(0, arg0);
+                        } else {
+                            arg0 = this.ensureArrayElementsOfType(arg0, arg1ElemType);
+                            ops.set(0, arg0);
+                        }
+                    } else if (arg0ElemType.mayBeNull && !arg1ElemType.mayBeNull) {
+                        if (arg1Vec.mayBeNull && arg1.equals(arg1Vec.nullValue())) {
+                            arg1 = new DBSPTypeVec(arg0ElemType, true).nullValue();
+                            ops.set(1, arg1);
+                        }
+                        arg1 = this.ensureArrayElementsOfType(arg1, arg0ElemType);
+                        ops.set(1, arg1);
+                    }
+                    // if the nullability is not the problem, return an unimplemented error as types are different
+                    else {
+                        throw new UnimplementedException(node);
+                    }
```

Comment:
It'd be nicer if Calcite simplified the array types to both be the same.
Here we try to simplify things a bit instead of having `4!` implementations of the same function based on the nullability.